src/main/java/com/direwolf20/buildinggadgets/common/integration/IBlockProviderItemHandler.java

```diff
@@ -0,0 +1,26 @@
+package com.direwolf20.buildinggadgets.common.integration;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+public interface IBlockProviderItemHandler extends IItemHandler {
+
+	/**
+	 * Returns the amount of a certain ItemStack that this IItemHandler currently has
```

Comment:
```suggestion
	 * Returns the amount of a certain {@link ItemStack} that this {@link IItemHandler} currently has without modifying it.
```

Reply:
I agree, the wording is a bit confusing. Additionally, we could say that it returns the amount which *can actually be removed*, to solve issues like Locked Slots.